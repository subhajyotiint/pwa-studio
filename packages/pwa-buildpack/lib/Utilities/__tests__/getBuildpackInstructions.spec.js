const getBuildpackInstructions = require('../getBuildpackInstructions');
const fse = require('fs-extra');
const { resolve } = require('path');

test('gets and loads an instruction from node modules', () => {
    expect(
        getBuildpackInstructions(
            'package-with-instruction',
            ['return-fse-readjsonsync']
        ).instructions['return-fse-readjsonsync']
    ).toBe(fse.readJSONSync);
});

test('gets and loads an instruction from any folder', () => {
    expect(
        getBuildpackInstructions(
            resolve(__dirname, '__fixtures__', 'non-package-folder-with-instruction'),
            ['return-fse-readjsonsync']
        ).instructions['return-fse-readjsonsync']
    ).toBe(fse.readJSONSync);
});

test('throws informative error if instruction fails', () => {
    expect(
        () => getBuildpackInstructions(
            'package-with-bad-instruction',
            ['throw-on-execute']
        )
    ).toThrowErrorMatchingSnapshot();
});

test('throws informative error if instruction does not exist', () => {
    expect(
        () => getBuildpackInstructions(
            'package-with-no-instructions',
            ['return-fse-readjsonsync']
        )
    ).toThrowErrorMatchingSnapshot();
});
