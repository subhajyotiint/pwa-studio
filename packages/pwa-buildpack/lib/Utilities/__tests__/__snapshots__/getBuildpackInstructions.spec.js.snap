// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws informative error if instruction does not exist 1`] = `"Buildpack could not find a valid '.buildpack/return-fse-readjsonsync.js' file in /Volumes/CaseSensitive/pwa-studio/packages/pwa-buildpack/lib/Utilities/__tests__/__fixtures__/modules/package-with-no-instructions. This file must be present to instruct Buildpack how to use the /Volumes/CaseSensitive/pwa-studio/packages/pwa-buildpack/lib/Utilities/__tests__/__fixtures__/modules/package-with-no-instructions package to perform the \\"return-fse-readjsonsync\\" operation. Original error: Cannot find module '/Volumes/CaseSensitive/pwa-studio/packages/pwa-buildpack/lib/Utilities/__tests__/__fixtures__/modules/package-with-no-instructions/.buildpack/return-fse-readjsonsync' from 'getBuildpackInstructions.js'"`;

exports[`throws informative error if instruction fails 1`] = `"Buildpack encountered an error running /Volumes/CaseSensitive/pwa-studio/packages/pwa-buildpack/lib/Utilities/__tests__/__fixtures__/modules/package-with-bad-instruction/.buildpack/throw-on-execute. Original error: Failed to run a bad instruction"`;
